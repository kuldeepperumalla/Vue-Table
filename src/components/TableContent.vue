<template>
  <div class="scroll">
    <h1>Table Data</h1>
    <v-simple-table>
      <template v-slot:default>
        <thead>
          <tr>
            <th class="text-left">Name</th>
            <th class="text-left">Contact</th>
            <th class="text-left">Id</th>
          </tr>
        </thead>
      </template>
    </v-simple-table>

    <v-responsive class="overflow-y-auto" max-height="500">
      <v-lazy
        :options="{
          threshold: .5
        }"
        min-height="500"
        transition="fade-transition"
      >
        <v-card>
          <div>
            <v-simple-table>
              <template v-slot:default>
                <tbody>
                  <tr
                    :key="post.Id"
                    v-for="(post) in posts"
                    :class="{highlight:post.removeSelectedItem}"
                    @dblclick="$set(post, 'removeSelectedItem', !post.removeSelectedItem)"
                  >
                    <td>
                      <input type="text" placeholder="Add name.." v-model="post.name" />
                    </td>
                    <td>
                      <input type="text" placeholder="Add contact.." v-model="post.contact" />
                    </td>
                    <td>
                      <input type="text" placeholder="Add Id.." v-model="post.id" />
                    </td>
                    <!-- <td>
                      <v-checkbox class="large" large v-model="post.removeCheckedItem" />
                    </td>-->
                  </tr>
                </tbody>
              </template>
            </v-simple-table>
          </div>
        </v-card>
      </v-lazy>
    </v-responsive>
  </div>
</template>




<script lang="ts">
import { Component, Vue, Prop } from "vue-property-decorator";

@Component
export default class TableContent extends Vue {
  @Prop({ required: true }) posts!: null;
}
</script>

<style>
tr.highlight {
  background: lightcoral;
}

h2 {
  margin-bottom: 0px;
}
table {
  font-family: arial, sans-serif;
  border-collapse: collapse;
  width: 100%;
}
td:hover {
  background: lightblue;
}
</style>


