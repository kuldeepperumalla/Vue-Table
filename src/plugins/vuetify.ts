import Vue from 'vue';
import vuetify from 'vuetify/lib';

Vue.use(vuetify, {});

export default new vuetify({
});
