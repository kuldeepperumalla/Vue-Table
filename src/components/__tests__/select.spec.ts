import { shallowMount, createLocalVue } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import vuetify from 'vuetify';
import TableContent from '../../components/TableContent.vue'




const localVue = createLocalVue()


describe('TableContent.vue', () => {

    beforeEach(() => {
        const localVue = createLocalVue()
        const wrapper = shallowMount(TableContent, {
            vuetify,
            localVue,
            propsData: {
                posts: 'posts'
            }
        })
    });






    test('should render content correctly', () => {
        const wrapper = shallowMount(TableContent, {
            vuetify,
            localVue
        });
        // expect(wrapper.vm.$el).toMatchSnapshot();
    });

    test('Selects the row item on click', () => {
        const wrapper = shallowMount(TableContent, {
            localVue,
            vuetify,


        });
        const classSelector = wrapper.find({ name: 'highlight' })
        // classSelector.trigger('click')
        expect(wrapper.contains('highlight')).toBe(false)

    });

    test('checks for class', () => {
        const wrapper = shallowMount(TableContent, {
            vuetify,
            localVue,

        })
        expect(wrapper.html()).toContain('class="scroll"')
    });



})