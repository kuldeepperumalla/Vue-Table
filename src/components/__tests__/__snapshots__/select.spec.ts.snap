// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableContent.vue should render content correctly 1`] = `
<div
  class="scroll"
>
  <h1>
    Table Data
  </h1>
   
  <v-simple-table />
   
  <v-responsive
    class="overflow-y-auto"
    max-height="500"
  >
    <v-lazy
      min-height="500"
      options="[object Object]"
      transition="fade-transition"
    >
      <v-card>
        <div>
          <v-simple-table />
        </div>
      </v-card>
    </v-lazy>
  </v-responsive>
</div>
`;
